### PR TITLE
Object Manipulator now maintains velocity on release

### PR DESCRIPTION
**Pull request by RogPodge**
## Overview
Fixed Object Manipulator behavior to maintain velocity on release

This fix also entailed fixing the GenericXRSDKController to properly read Velocity and Angular Velocity Data from the input devices. However, this also required a fix from the XRSDK side of things, so this PR combined with the to be completed update to the Windows XRSDK plugin will solve this issue completely

Additionally, I found that objects that are thrown with the controller's velocity feel a bit dampened and unnaturally stiff compared to letting them maintain their velocity. I have included examples of both behavior in the video clips below, which looks the best/most natural?

Old Behavior:
https://user-images.githubusercontent.com/39840334/116744273-f3864a00-a9ae-11eb-90d8-4e99e120f932.mp4

Controller dampened velocity:
https://user-images.githubusercontent.com/39840334/116744492-37794f00-a9af-11eb-9016-6989f4e23834.mp4

Maintaining original velocity:
https://user-images.githubusercontent.com/39840334/116744514-3e07c680-a9af-11eb-93b4-6cea3eacb7dc.mp4

## Changes
- Fixes: #9690

### CODE REVIEW

File: Assets/MRTK/Providers/XRSDK/Controllers/GenericXRSDKController.cs [hunk 1]
```diff
@@ -137,6 +116,63 @@ protected virtual void UpdateSixDofData(InputDevice inputDevice)
...
+            }
+        }
+
+        private static readonly ProfilerMarker UpdateVelocityPerfMarker = new ProfilerMarker("[MRTK] GenericXRSDKController.UpdateVelocity");
+
+        public void UpdateVelocity(InputDevice inputDevice)
+        {
+            using (UpdateVelocityPerfMarker.Auto())
+            {
+                Debug.Log("updating velocity");
```
Comment (keveleigh) [inline Assets/MRTK/Providers/XRSDK/Controllers/GenericXRSDKController.cs:161]: Can this be removed?

File: Assets/MRTK/Providers/XRSDK/Controllers/GenericXRSDKController.cs [hunk 2]
```diff
@@ -137,6 +116,63 @@ protected virtual void UpdateSixDofData(InputDevice inputDevice)
...
+                Vector3 newVelocity;
+                if (inputDevice.TryGetFeatureValue(CommonUsages.deviceVelocity, out newVelocity))
+                {
+                    Debug.Log("updating velocity2" + newVelocity);
```
Comment (keveleigh) [inline Assets/MRTK/Providers/XRSDK/Controllers/GenericXRSDKController.cs:165]: This one too?

File: Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs [hunk 3]
```diff
@@ -973,16 +973,6 @@ private void ReleaseRigidBody(Vector3 velocity, Vector3 angularVelocity)
             {
                 rigidBody.useGravity = wasGravity;
                 rigidBody.isKinematic = wasKinematic;
-
-                if (releaseBehavior.HasFlag(ReleaseBehaviorType.KeepVelocity))
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs:977]: Are these flags used elsewhere anymore, or do we basically ignoring them now?

Comment (RogPodge) [reply Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs:977]: yeah they're only used here

I can get rid of the variable if we decide to go with option 3's behavior

Comment (RogPodge) [reply Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs:977]: See comment in the main thread

**Comment by RogPodge** [comment]
We've settled on maintaining the original behavior (matching controller velocities) on near interactions only, while on far interactions we allow the object to keep it's own velocities. 

The reason for dividing the behavior into these two cases is because at far ranges, the velocity of the controller no longer matches the velocity of the object, since moving your hand a short distance ends up moving the manipulated object by a much larger distance. This makes it so that the thrown object has a unnatural dampened velocity. Casing on the different kinds of interactions will let us get more natural movement for each interaction

File: Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs [hunk 4]
```diff
@@ -973,7 +973,12 @@ private void ReleaseRigidBody(Vector3 velocity, Vector3 angularVelocity)
             {
                 rigidBody.useGravity = wasGravity;
                 rigidBody.isKinematic = wasKinematic;
+            }
 
+            // Match the object't velocity to the controller for near interactions
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs:978]:
```suggestion
            // Match the object's velocity to the controller for near interactions
```

File: Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs [hunk 5]
```diff
@@ -973,7 +973,12 @@ private void ReleaseRigidBody(Vector3 velocity, Vector3 angularVelocity)
             {
                 rigidBody.useGravity = wasGravity;
                 rigidBody.isKinematic = wasKinematic;
+            }
```
Comment (keveleigh) [inline Assets/MRTK/SDK/Features/Input/Handlers/ObjectManipulator.cs:976]: I think the blow code needs to stay in this `if` block, so we don't try to manipulate a null `rigidBody`

**Review by keveleigh — APPROVED** [review]

**Pull request merged**